lib/node-package-manager.ts

```diff
@@ -8,139 +7,99 @@ interface INpmOpts {
 }
 
 export class NodePackageManager implements INodePackageManager {
-	constructor(private $childProcess: IChildProcess,
+	constructor(private $fs: IFileSystem,
+		private $errors: IErrors,
+		private $childProcess: IChildProcess,
 		private $logger: ILogger,
 		private $options: IOptions) { }
 
-	public getCache(): string {
-		return npm.cache;
-	}
-
-	public load(config?: any): IFuture<void> {
-		if (npm.config.loaded) {
-			let data = npm.config.sources.cli.data;
-			Object.keys(data).forEach(k => delete data[k]);
-			if (config) {
-				_.assign(data, config);
-			}
-			return Future.fromResult();
-		} else {
-			let future = new Future<void>();
-			npm.load(config, (err: Error) => {
-				if (err) {
-					future.throw(err);
-				} else {
-					future.return();
-				}
-			});
-			return future;
+	public install(packageName: string, pathToSave: string, config?: any): any {
+		if (this.$options.disableNpmInstall) {
+			return;
+		}
+		if (this.$options.ignoreScripts) {
+			config = config || {};
+			config["ignore-scripts"] = true;
 		}
-	}
 
-	public install(packageName: string, pathToSave: string, config?: any): IFuture<any> {
-		return (() => {
-			if (this.$options.disableNpmInstall) {
-				return;
-			}
-			if (this.$options.ignoreScripts) {
-				config = config || {};
-				config["ignore-scripts"] = true;
+		try {
+			let subDirsBefore:Array<string> = this.$fs.exists(path.join(pathToSave, "node_modules")).wait() ? this.$fs.readDirectory(path.join(pathToSave, "node_modules")).wait() : [];
+
+			let flags = this.getFlagsString(config);
+			this.$childProcess.exec(`npm install ${packageName} ${flags}`, { cwd: pathToSave }).wait();
```

Comment:
This `exec` invocation looks like it's eating the child process stdout. I think we should inherit stdio here so that npm-related errors are visible to the user. Hmm, yes, we need stdin too -- I've heard rumors on the internet there are packages that block the console asking for email addresses on installation.